### PR TITLE
Port encapsulate field tests to new harness

### LINKED ISSUE #35701: AccessViolationException in IVsPreviewChangesService.PreviewChanges
**Version Used**: Integration test run for #35603 

```
Event Time (UTC): 5/14/2019 5:47:31 PM
Event ID: 1026
Data: Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.AccessViolationException
   at Microsoft.VisualStudio.Shell.Interop.IVsPreviewChangesService.PreviewChanges(Microsoft.VisualStudio.Shell.Interop.IVsPreviewChangesEngine)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Preview.PreviewDialogService.PreviewChanges(System.String, System.String, System.String, System.String, Microsoft.CodeAnalysis.Glyph, Microsoft.CodeAnalysis.Solution, Microsoft.CodeAnalysis.Solution, Boolean)
   at Microsoft.CodeAnalysis.Editor.Implementation.EncapsulateField.AbstractEncapsulateFieldCommandHandler.Execute(Microsoft.VisualStudio.Text.Editor.Commanding.Commands.EncapsulateFieldCommandArgs, Microsoft.VisualStudio.Utilities.IUIThreadOperationScope)
   at Microsoft.CodeAnalysis.Editor.Implementation.EncapsulateField.AbstractEncapsulateFieldCommandHandler.ExecuteCommand(Microsoft.VisualStudio.Text.Editor.Commanding.Commands.EncapsulateFieldCommandArgs, Microsoft.VisualStudio.Commanding.CommandExecutionContext)
   at Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](Microsoft.VisualStudio.Commanding.ICommandHandler, System.__Canon, System.Action, Microsoft.VisualStudio.Commanding.CommandExecutionContext)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService+<>c__DisplayClass13_1`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<Execute>b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(System.Object, System.Action, System.Predicate`1<System.Exception>)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService+<>c__DisplayClass13_1`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<Execute>b__0()
   at Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](Microsoft.VisualStudio.Commanding.ICommandHandler, System.__Canon, System.Action, Microsoft.VisualStudio.Commanding.CommandExecutionContext)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService+<>c__DisplayClass13_1`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<Execute>b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(System.Object, System.Action, System.Predicate`1<System.Exception>)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService+<>c__DisplayClass13_1`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<Execute>b__0()
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.ExecuteCommandHandlerChain(Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerServiceState, System.Action, System.Action)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.Execute[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Func`3<Microsoft.VisualStudio.Text.Editor.ITextView,Microsoft.VisualStudio.Text.ITextBuffer,System.__Canon>, System.Action)
   at Microsoft.VisualStudio.Editor.Implementation.CommandHandlerServiceAdapter.ExecuteEncapsulateFieldCommand(System.Action)
   at Microsoft.VisualStudio.Editor.Implementation.CommandHandlerServiceAdapter.ExecuteVisualStudio2000(UInt32, IntPtr, System.Action)
   at Microsoft.VisualStudio.Editor.Implementation.CommandHandlerServiceAdapter.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandHandlerServiceFilter.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.LanguageServices.Implementation.AbstractOleCommandTarget+<>c__DisplayClass36_0.<ExecuteVisualStudio2000>g__executeNextCommandTarget|0()
   at Microsoft.CodeAnalysis.Editor.Implementation.Commands.CommandHandlerService.ExecuteHandlers[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Generic.IList`1<Microsoft.CodeAnalysis.Editor.ICommandHandler`1<System.__Canon>>, System.__Canon, System.Action)
   at Microsoft.CodeAnalysis.Editor.Implementation.Commands.CommandHandlerService.Microsoft.CodeAnalysis.Editor.ICommandHandlerService.Execute[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](Microsoft.VisualStudio.Utilities.IContentType, System.__Canon, System.Action)
   at Microsoft.VisualStudio.LanguageServices.Implementation.AbstractOleCommandTarget.ExecuteEncapsulateField(Microsoft.VisualStudio.Text.ITextBuffer, Microsoft.VisualStudio.Utilities.IContentType, System.Action)
   at Microsoft.VisualStudio.LanguageServices.Implementation.AbstractOleCommandTarget.ExecuteVisualStudio2000(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr, Microsoft.VisualStudio.Text.ITextBuffer, Microsoft.VisualStudio.Utilities.IContentType)
   at Microsoft.VisualStudio.LanguageServices.Implementation.AbstractOleCommandTarget.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandFilterWrapper.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandFilterWrapper.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.Intellisense.MultiCaretPreemptingFilterProvider+MultiCaretPreemptingFilter.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.SimpleTextViewWindow.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Editor.Implementation.CompoundTextViewWindow.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Platform.WindowManagement.DocumentObjectSite.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
   at Microsoft.VisualStudio.Platform.WindowManagement.WindowFrame.Exec(System.Guid ByRef, UInt32, UInt32, IntPtr, IntPtr)
```

### PR DESCRIPTION
**Pull request by sharwell**
This test was observed in a recent failing run on roslyn-integration-CI due to use of light bulb in the old integration test harness.

Also update State Reset to automatically dismiss dialogs remaining from a previous test (e.g. the Save File dialog).

Closes #35701

### CODE REVIEW

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/DialogHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+{
+    public static class DialogHelpers
+    {
+        /// <summary>
+        /// Returns an <see cref="IUIAutomationElement"/> representing the open dialog with automation ID
+        /// <paramref name="dialogAutomationId"/>.
+        /// Throws an <see cref="InvalidOperationException"/> if an open dialog with that name cannot be
+        /// found.
+        /// </summary>
+        public static async Task<IUIAutomationElement> GetOpenDialogByIdAsync(JoinableTaskFactory joinableTaskFactory, IntPtr visualStudioHWnd, string dialogAutomationId, CancellationToken cancellationToken)
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/DialogHelpers.cs:26]: 📝 We should switch to WPF APIs for dialogs we own, and only use UI Automation for dialogs that might not be in WPF. That can be done later though.

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InfrastructureTests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                @"
+namespace MyNamespace
+{
+$$
+}",
+                HangMitigatingCancellationToken);
+
+            // Trigger a call to File.Close to ensure we can recover from it
+            await TestServices.Input.SendAsync(new KeyPress(VirtualKey.F, ShiftState.Alt), VirtualKey.C);
+            //await Task.Delay(TimeSpan.FromSeconds(2));
```
Comment (allisonchou) [inline src/VisualStudio/IntegrationTest/New.IntegrationTests/InfrastructureTests.cs:37]: Is this leftover code from debugging?

**Review by allisonchou — APPROVED** [review]
🎉

File: src/VisualStudio/IntegrationTest/New.IntegrationTests/InfrastructureTests.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/IntegrationTest/New.IntegrationTests/InfrastructureTests.cs]: Yes, now removed.

**Pull request merged**